### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Closes: #800

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted `dev` branch
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all tests passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by blushi — APPROVED** [review]

File: server/routes/files.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import * as express from 'express';
```
Comment (blushi) [inline server/routes/files.ts:1]: could you fix the missing `;` in this file? (it's ok to push directly to `dev` for such cosmetic changes)
we should probably add some linting in this repo, I'll create an issue for that

File: server/routes/files.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import * as bodyParser from 'body-parser';
+const S3 = require('aws-sdk/clients/s3')
+const { Readable } = require('stream');
+
+const bucketName = process.env.AWS_S3_BUCKET
+const region = process.env.AWS_BUCKET_REGION
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY
```
Comment (blushi) [inline server/routes/files.ts:9]: please make sure we are not missing any variables in server/.env.example

Comment (mhagel) [reply server/routes/files.ts:9]: Ok, also, the AWS key is changed to allow the new update policy. This is a reminder to myself to update that in PROD.

File: server/routes/files.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import * as express from 'express';
+import * as bodyParser from 'body-parser';
+const S3 = require('aws-sdk/clients/s3')
```
Comment (mhagel) [inline server/routes/files.ts:3]:
```suggestion
const S3 = require('aws-sdk/clients/s3');
```

File: server/routes/files.ts (hunk 2 above)
Comment (mhagel) [inline server/routes/files.ts:9]:
```suggestion
const bucketName = process.env.AWS_S3_BUCKET;
const region = process.env.AWS_BUCKET_REGION;
const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
```

File: server/routes/files.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const bucketName = process.env.AWS_S3_BUCKET
+const region = process.env.AWS_BUCKET_REGION
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY
+
+const s3 = new S3({
+  region,
+  accessKeyId,
+  secretAccessKey
+})
```
Comment (mhagel) [inline server/routes/files.ts:15]:
```suggestion
});
```

File: server/routes/files.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    Bucket: `${bucketName}/projects/${projectId}`,
+    Key: key,
+  };
+
+  s3.deleteObject(deleteParams, function (err, data) {
+    if (err) {
+      console.log('s3 Error', err);
+      response.status(500).send(err);
+    } if (data) {
+      response.send('File successfully deleted')
```
Comment (mhagel) [inline server/routes/files.ts:67]:
```suggestion
      response.send({imageUrl: data.Location});
    }
  });
});

router.delete('/files/:projectId/:key', bodyParser.json(), (request, response: express.Response) => {
  const projectId = request.params.projectId;
  const key = request.params.key;

  const deleteParams = {
    Bucket: `${bucketName}/projects/${projectId}`,
    Key: key,
  };

  s3.deleteObject(deleteParams, function (err, data) {
    if (err) {
      console.log('s3 Error', err);
      response.status(500).send(err);
    } if (data) {
      response.send('File successfully deleted');
```

File: server/routes/files.ts (hunk 2 above)
Comment (mhagel) [reply server/routes/files.ts:9]: just need to add `AWS_BUCKET_REGION=`

File: server/routes/files.ts (hunk 1 above)
Comment (mhagel) [reply server/routes/files.ts:1]: made a new PR to dev, if that is easiest: https://github.com/regen-network/registry-server/pull/105

**Review by haveanicedavid — APPROVED** [review]

File: server/routes/files.ts (hunk 1 above)
Comment (blushi) [reply server/routes/files.ts:1]: approved and merged into `dev`

**Pull request merged**